mm2src/mm2_bitcoin/chain/src/transaction.rs

```diff
@@ -562,21 +562,32 @@ impl Deserializable for Transaction {
         Self: Sized,
         T: io::Read,
     {
-        // read the entire buffer to get it's copy for different cases
-        // it works properly only when buffer contains only 1 transaction bytes
-        // it breaks block serialization, but block serialization is not required for AtomicDEX
-        // specific use case
+        // Read the entire buffer in 4KB chunks.
+        //
+        // Assumes a single transaction per buffer; not designed for block serialization,
+        // which is unnecessary in komodo-defi-framework use case.
         let mut buffer = vec![];
-        reader.read_to_end(&mut buffer)?;
```

Comment:
I wonder why you do not want to use read_to_end().
I took a look at its code, apparently it uses a too small buffer of 32 byte to read, is this the reason?